File: nodejs/src/commands/auth/signup.ts
import { Command } from 'commander';
import inquirer from 'inquirer';
import chalk from 'chalk';
import { AuthService } from '../../services/auth';
import { validateEmail, validatePassword } from '../../utils/validators';

export const signupCommand = new Command()
  .name('signup')
  .description('Create a new Elitecode account')
  .action(async () => {
    try {
      const answers = await inquirer.prompt([
        {
          type: 'input',
          name: 'name',
          message: 'Name:',
          validate: (input) => input.length > 0 || 'Name is required'
        },
        {
          type: 'input',
          name: 'username',
          message: 'Username:',
          validate: (input) => input.length >= 3 || 'Username must be at least 3 characters'
        },
        {
          type: 'input',
          name: 'email',
          message: 'Email:',
          validate: validateEmail
        },
        {
          type: 'password',
          name: 'password',
          message: 'Password:',
          validate: validatePassword
        },
        {
          type: 'password',
          name: 'confirmPassword',
          message: 'Confirm Password:',
          validate: (input, answers) => {
            if (input !== answers.password) {
              return 'Passwords do not match';
            }
            return true;
          }
        }
      ]);

      const authService = new AuthService();
      const result = await authService.signup(answers);
      
      console.log(chalk.green('✅ Account created successfully!'));
      console.log(chalk.blue(`Welcome, ${result.user.name}!`));
      
    } catch (error) {
      console.error(chalk.red('❌ Signup failed:', error.message));
    }
  });

File: cmd/auth/signup.go
package auth

import (
	"fmt"
	"strings"

	"github.com/AlecAivazis/survey/v2"
	"github.com/fatih/color"
	"github.com/spf13/cobra"
	"github.com/IshaanNene/EliteCode-brew/internal/api"
	"github.com/IshaanNene/EliteCode-brew/internal/storage"
)

var signupCmd = &cobra.Command{
	Use:   "signup",
	Short: "Create a new Elitecode account",
	RunE:  runSignup,
}

func runSignup(cmd *cobra.Command, args []string) error {
	var credentials struct {
		Name            string
		Username        string
		Email           string
		Password        string
		ConfirmPassword string
	}

	questions := []*survey.Question{
		{
			Name: "name",
			Prompt: &survey.Input{
				Message: "Full Name:",
			},
			Validate: survey.Required,
		},
		{
			Name: "username",
			Prompt: &survey.Input{
				Message: "Username:",
			},
			Validate: survey.Required,
		},
		{
			Name: "email",
			Prompt: &survey.Input{
				Message: "Email:",
			},
			Validate: survey.Required,
		},
		{
			Name: "password",
			Prompt: &survey.Password{
				Message: "Password:",
			},
			Validate: survey.Required,
		},
		{
			Name: "confirmpassword",
			Prompt: &survey.Password{
				Message: "Confirm Password:",
			},
			Validate: survey.Required,
		},
	}

	err := survey.Ask(questions, &credentials)
	if err != nil {
		return err
	}

	// Validate passwords match
	if credentials.Password != credentials.ConfirmPassword {
		return fmt.Errorf("passwords do not match")
	}

	// Validate email format
	if !strings.Contains(credentials.Email, "@") {
		return fmt.Errorf("invalid email format")
	}

	client := api.NewClient()
	user, token, err := client.Signup(credentials.Name, credentials.Username, credentials.Email, credentials.Password)
	if err != nil {
		return fmt.Errorf("signup failed: %v", err)
	}

	// Store auth token and user info
	config := storage.NewConfigManager()
	if err := config.SetAuthToken(token); err != nil {
		return fmt.Errorf("failed to store auth token: %v", err)
	}

	if err := config.SetUser(user); err != nil {
		return fmt.Errorf("failed to store user info: %v", err)
	}

	color.Green("✅ Account created successfully!")
	color.Blue("Welcome to Elitecode, %s!", user.Name)
	
	return nil
}